### DISCUSSION TITLE
batch resizing thousands of  photos from multiple folders

### DISCUSSION THREAD

**Discussion by junomarian** [ImageMagick/ImageMagick#4954] [Contribute]
I have various folders with high res images sizes some more than 6000px x 4000px. Potrait as well landscape. some are 5000px x 3000px, size varies with folder to folder. I want to resize all images for saving resources like loading time/processing/memory space. I want to keep all images around 3000x2000. How I can do this with imagemagick? in batch! Replacing original with same name.

Earlier i used photosizer software on windows. Selecting folder one by one, then process images with jpegMini software. So 5 mb photos decrease to the size to around 1.5 mb. Every folder contains images with different resolution. Its a long time consuming process.

**Comment by snibgo** [comment]
To help you, we need to know what version of IM you use, and on what platform.

**Reply by junomarian** [reply]
`$ convert -version
Version: ImageMagick 7.1.0-16 Q16-HDRI aarch64 2021-11-20 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI
Delegates (built-in): bzlib cairo fftw fontconfig freetype heic jng jp2 jpeg jxl lcms lzma pangocairo png rsvg tiff webp x xml zlib
Compiler: gcc (4.2)`

I am on Android Galaxy Tab S7 FE .

**Comment by fmw42** [comment]
You can use mogrify to process a folder of images in one command. If you have more than one folder or multiple folders in a given folder, you will have to run mogrify once for each folder. See 

https://legacy.imagemagick.org/Usage/basics/#mogrify

For ImageMagick 7, use "magick mogrify" in place of "mogrify"

**Comment by snibgo** [comment]
I suggest the first step is to decide on the IM command that does what you want. Something like this:
```
magick in.ext -auto-orient -resize 3000x2000 outdir/out.ext
```
That assumes your output directory already exists. Instead of in.ext and out.ext, use your real filenames.

You can adapt that to a "magick mogrify" command to do all the files in one directory, then put that in a shell (bash) loop to loop through the subdirectories. I don't do much bash, so can't help you there.

EDIT: Wrote "magick montage" but meant "magick mogrify".

**Reply by fmw42** [reply]
I believe he meant magick mogrify not magick montage.

**Comment by snibgo** [comment]
Oops, yes, thanks. Now corrected.
